python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -55,75 +64,249 @@ def __init__(
-        if not isinstance(value, GeoColumn):
-            raise TypeError
-        self._column = value
+        @property
+        def x(self):
+            return cudf.Series(self._col.leaves().values[0::2])
+
+        @property
+        def y(self):
+            return cudf.Series(self._col.leaves().values[1::2])
+
+        @property
+        def xy(self):
+            return cudf.Series(self._col.leaves().values)
 
     @property
     def points(self):
         """
         Access the `PointsArray` of the underlying `GeoArrowBuffers`.
         """
-        return self._geocolumn.points
+        return self.GeoColumnAccessor(self._column.points)
 
     @property
     def multipoints(self):
         """
         Access the `MultiPointArray` of the underlying `GeoArrowBuffers`.
         """
-        return self._geocolumn.multipoints
+        return self.GeoColumnAccessor(self._column.mpoints)
 
     @property
     def lines(self):
         """
         Access the `LineArray` of the underlying `GeoArrowBuffers`.
         """
-        return self._geocolumn.lines
+        return self.GeoColumnAccessor(self._column.lines)
 
     @property
     def polygons(self):
         """
         Access the `PolygonArray` of the underlying `GeoArrowBuffers`.
         """
-        return self._geocolumn.polygons
+        return self.GeoColumnAccessor(self._column.polygons)
 
     def __repr__(self):
         # TODO: Implement Iloc with slices so that we can use `Series.__repr__`
         return self.to_pandas().__repr__()
 
+    class GeoSeriesLocIndexer:
+        """
+        Not yet supported.
+        """
+
+        def __init__(self):
+            # Todo: Easy to implement with a join.
+            raise NotImplementedError
+
+    class GeoSeriesILocIndexer:
+
+        """
+        Each row of a GeoSeries is one of the six types: Point, MultiPoint,
+        LineString, MultiLineString, Polygon, or MultiPolygon.
+        """
+
+        def __init__(self, sr):
+            self._sr = sr
+
+        @cached_property
+        def _type_int_to_field(self):
+            return {
+                Feature_Enum.POINT: self._sr.points,
+                Feature_Enum.MULTIPOINT: self._sr.mpoints,
+                Feature_Enum.LINESTRING: self._sr.lines,
+                Feature_Enum.MULTILINESTRING: self._sr.lines,
+                Feature_Enum.POLYGON: self._sr.polygons,
+                Feature_Enum.MULTIPOLYGON: self._sr.polygons,
+            }
+
+        @cached_property
+        def _get_shapely_class_for_Feature_Enum(self):
+            type_map = {
+                Feature_Enum.POINT: Point,
+                Feature_Enum.MULTIPOINT: MultiPoint,
+                Feature_Enum.LINESTRING: LineString,
+                Feature_Enum.MULTILINESTRING: MultiLineString,
+                Feature_Enum.POLYGON: Polygon,
+                Feature_Enum.MULTIPOLYGON: MultiPolygon,
+            }
+            return type_map
+
+        def __getitem__(self, item):
```

Comment:
I think this part can use some (future) optimizations. The major part of doing slicing on host is quite inefficient and we can probably rewrite them by slicing the underlying children since they are all cudf Series now.